includes/actions.php

```diff
@@ -33,6 +33,24 @@ function wpum_delete_pages_transient( $post_id ) {
 }
 add_action( 'save_post_page', 'wpum_delete_pages_transient' );
 
+/**
+ * Delete cached list of pages when a page is deleted.
+ *
+ * @param int $post_id
+ * @param WP_Post $post
+ * @return void
+ */
+function wpum_delete_pages_transient_on_delete( $post_id, $post ) {
+
+	if ( wp_is_post_revision( $post_id ) || $post->post_type !== 'page' ) {
+		return;
+	}
+
+	delete_transient( 'wpum_get_pages' );
+
+}
+add_action( 'delete_post', 'wpum_delete_pages_transient_on_delete', 99, 2 );
```

Comment:
@achchu93 need to also bust the transient when a new page is added too

Reply:
@polevaultweb It is already there. Check the line 25-34